platform/srf06-cc26xx/sensortag/mpu-9250-sensor.c

```diff
@@ -377,6 +432,121 @@ enable_sensor(uint16_t axes)
   }
 }
 /*---------------------------------------------------------------------------*/
+/**
+ * \brief Enable the MAG
+ */
+static void 
+mag_enable(void)
+{
+  uint8_t val;
+  if(power_test() != 0) {
+    /* Set magnetometer configurtation, read mode and data resolution */
+    if(!set_bypass()) {
+      return;
+    }
+    SENSOR_SELECT_MAG();  
+    val = (scale << 4) | mode; 
+    sensor_common_write_reg(MAG_CNTL1, &val, 1);
+    delay_ms(10);
+    val = 0;
+    sensor_common_read_reg(MAG_WHO_AM_I, &val, 1);
+    
+  }
+  SENSOR_DESELECT();
+}
+/*---------------------------------------------------------------------------*/
+/**
+* \brief  Shift the I2C bus to control the magnetomoter
+* \return true if success
+*/
+static bool 
+set_bypass(void)
+{
+  bool success;
+  uint8_t val;
+
+  SENSOR_SELECT();
+  delay_ms(100);
+  val = EN_BYPASS;
+  sensor_common_write_reg(INT_PIN_CFG, &val, 1);
+  delay_ms(10);
+  val = 0;
+  sensor_common_read_reg(INT_PIN_CFG, &val, 1);
+  SENSOR_DESELECT();
+  if(val == 0x22) {
+    success = true;
+  }
+  else {
+    success = false;
+  }
+  return success;
```

Comment:
The success/failure of the function should be returned directly:
```c
  return val == 0x22;
```

Reply:
Thank you, i have implemented that for the next commit